.github/workflows/default.yaml

```diff
@@ -146,7 +157,7 @@ jobs:
         if: success() || failure()
         uses: actions/upload-artifact@v3
         with:
-          name: build-logs-${{ runner.os }}
+          name: build-logs-${{ matrix.os }}-${{ matrix.compiler }}
```

Comment:
Using runner.os here is just logically wrong.
runner.os is Linux for ubuntu-22.04, ubuntu-20.04, freebsd (which runs on ubuntu-latest), openbsd (ditto) and who knows what else. We need to add matrix.compiler here, otherwise logs for clang and gcc will clash and overwrite each other. Mix-n-match of runner.os and matrix.compiler makes no sense at all.

The alternative is to run a PR to change "runner" to "matrix" standalone, and wait for things to settle and deal with merge conflicts.

Reply:
In other words: if we want to progressively replace jenkins with github workflows, we can't avoid this change. If it belongs to this PR or to another PR, is purely a matter of convenience

Reply:
>> If there is a good reason for doing this change here and now, please disclose that reason in PR description.

> We need to add matrix.compiler here, otherwise logs for clang and gcc will clash and overwrite each other. 

Please add that info to the PR description. Adding layers to the matrix in #1787 did not result in clashes [AFAICT](https://github.com/squid-cache/squid/pull/1787/files#r1571154576), but I was not sure whether GitHub is smart enough to handle clashes when adding compilers. (I am aware of the difference in log file names between those two use cases.) Thank you for answering my question.

Reply:
> but I was not sure whether GitHub is smart enough to handle clashes when adding compilers

The key difference between these two cases is that in this cases, archived filenames are the same so there is no way to collate them.

Reply:
> > I am aware of the difference in log file names between those two use cases.

> The key difference between these two cases is that in this cases, archived filenames are the same so there is no way to collate them.

As I said, I am aware of this file naming difference between the two cases. I suspect there is a way to avoid multiplying the number of top-level artifacts, but, at this time, I am not suggesting that we spend time on that.